src/core/dyad_core.c

```diff
@@ -0,0 +1,781 @@
+static int dyad_commit(dyad_ctx_t *ctx, const char *fname)
+#else
+static inline int dyad_commit(dyad_ctx_t *ctx, const char *fname)
+#endif
+{
+    int rc = 0;
+    char upath[PATH_MAX] = {'\0'};
+    // If a context is not provided, print an error and return DYAD_NOCTX
+    if (ctx == NULL)
+    {
+        fprintf (stderr, "DYAD context not provided to dyad_commit\n");
+        return DYAD_NOCTX;
+    }
+    // Extract the path to the file specified by fname relative to the
+    // producer-managed path
+    // This relative path will be stored in upath
+    if (!cmp_canonical_path_prefix (ctx->prod_managed_path, fname, upath, PATH_MAX))
+    {
+        DYAD_LOG_INFO (ctx, "%s is not in the Producer's managed path\n", fname);
+        rc = DYAD_OK;
+        goto commit_done;
+    }
+    // Call publish_via_flux to actually store information about the file into
+    // the Flux KVS
+    // Fence this call with reassignments of reenter so that, if intercepting
+    // file I/O API calls, we will not get stuck in infinite recursion
+    ctx->reenter = false;
+    rc = publish_via_flux(ctx, upath);
+    ctx->reenter = true;
+
+commit_done:
+    // If "check" is set and the operation was successful, set the
+    // DYAD_CHECK_ENV environment variable to "ok"
+    if (rc == DYAD_OK && (ctx && ctx->check))
+    {
+        setenv(DYAD_CHECK_ENV, "ok", 1);
+    }
+    return rc;
+}
+
+#if DYAD_PERFFLOW
+__attribute__((annotate("@critical_path()")))
+static int dyad_kvs_lookup(dyad_ctx_t *ctx, const char* kvs_topic, uint32_t *owner_rank)
+#else
+static inline int dyad_kvs_lookup(dyad_ctx_t *ctx, const char* kvs_topic, uint32_t *owner_rank)
+#endif
+{
+    int rc = -1;
+    flux_future_t *f = NULL;
+    // Lookup information about the desired file (represented by kvs_topic)
+    // from the Flux KVS. If there is no information, wait for it to be
+    // made available
+    f = flux_kvs_lookup(ctx->h,
+            ctx->kvs_namespace,
+            FLUX_KVS_WAITCREATE,
+            kvs_topic
+            );
+    // If the KVS lookup failed, log an error and return DYAD_BADLOOKUP
+    if (f == NULL)
+    {
+        DYAD_LOG_ERR (ctx, "KVS lookup failed!\n");
+        return DYAD_BADLOOKUP;
+    }
+    // Extract the rank of the producer from the KVS response
+    rc = flux_kvs_lookup_get_unpack(f, "i", owner_rank);
+    // No matter what, destroy the Flux future if needed
+    if (f != NULL)
+    {
+        flux_future_destroy(f);
+        f = NULL;
+    }
+    // If the extraction did not work, log an error and return DYAD_BADFETCH
+    if (rc < 0)
+    {
+        DYAD_LOG_ERR(ctx, "Could not unpack owner's rank from KVS response\n");
+        return DYAD_BADFETCH;
+    }
+    return DYAD_OK;
+}
+
+#if DYAD_PERFFLOW
+__attribute__((annotate("@critical_path()")))
+static int dyad_fetch(dyad_ctx_t *ctx, const char* fname,
+        dyad_kvs_response_t **resp)
+#else
+static inline int dyad_fetch(dyad_ctx_t *ctx, const char* fname,
```

Comment:
`dyad_ctx_t *ctx` -> `const dyad_ctx_t *ctx`

Reply:
same for validate_context, dyad_kvs_commit, publish_via_flux, dyad_kvs_lookup, dyad_rpc_get, and dyad_pull